src/crispy/utils.h

```diff
@@ -440,6 +442,51 @@ inline std::optional<unsigned> fromHexDigit(char value)
     return std::nullopt;
 }
 
+inline std::string unescapeURL(std::string_view input)
+{
+    std::string result;
+    result.reserve(input.size());
+    for (size_t i = 0; i < input.size(); ++i)
+    {
+        if (input[i] == '%' && i + 2 < input.size())
+        {
+            auto const h1 = crispy::fromHexDigit(input[i + 1]);
+            auto const h2 = crispy::fromHexDigit(input[i + 2]);
+            if (h1.has_value() && h2.has_value())
+            {
+                result.push_back(static_cast<char>((h1.value() << 4) | h2.value()));
+                i += 2;
+                continue;
+            }
+        }
+        result.push_back(input[i]);
+    }
+    return result;
+}
+
+template <typename Callback>
+constexpr void for_each_key_value(std::string_view text,
```

Comment:
better restrict callback function using `std::is_invocable_v`
also, this two chars can be easily swapped by accident, can we use strong types here?

Reply:
done